src/ddmd/objc.d

```diff
@@ -117,126 +117,140 @@ struct ObjcSelector
     }
 }
 
-struct Objc_ClassDeclaration
-{
-    // true if this is an Objective-C class/interface
-    bool objc;
+private __gshared Objc _objc;
 
-    // MARK: Objc_ClassDeclaration
-    extern (C++) bool isInterface()
-    {
-        return objc;
-    }
+Objc objc()
+{
+    return _objc;
 }
 
-struct Objc_FuncDeclaration
+interface Objc
 {
-    FuncDeclaration fdecl;
-    // Objective-C method selector (member function only)
-    ObjcSelector* selector;
-
-    extern (D) this(FuncDeclaration fdecl)
+    static void _init()
     {
-        this.fdecl = fdecl;
+        if (global.params.isOSX && global.params.is64bit)
```

Comment:
This condition is OK for now, but I'd like to move it to somewhere else in future.  Maybe some `Compiler` interface, as `Target` doesn't quite fit.